src/Form.js

```diff
@@ -33,19 +33,24 @@ class Form extends Component {
     };
 
     initialValues = {};
+    transforms = {};
 
     onChange = e => {
         this.props.onChange(e);
 
-        if (!this.state.config[e.target.name]) {
+        const { checked, name, type, value } = e.target;
+
+        if (!this.state.config[name]) {
             return;
         }
 
+        const transform =
+            typeof this.transforms[name] === 'function'
+                ? this.transforms[name]
+                : value => value;
+
         this.setField({
-            [e.target.name]:
-                e.target.type === 'checkbox'
-                    ? e.target.checked
-                    : e.target.value,
+            [name]: transform(type === 'checkbox' ? checked : value),

```

Comment:
A stylistic thing, but it also avoids creating an arrow function when not required.

How about?

```js
let finalValue = type === 'checkbox' ? checked : value;
if (transform[e.target.name]) {
  finalValue = transform[e.target.name](value);
}
this.setField({ [name]: finalValue });
```